### DISCUSSION TITLE
Failed to establish connection to NetBox API

### DISCUSSION THREAD

**Discussion by ellismk** [netbox-community/netbox#9605] [Q&A] [answered]
### NetBox version

v3.2.5

### Python version

3.9

### Steps to Reproduce

Environment: Sandbox closed
Rocky Linux 3.8
ansible 2.12.2 core
python 3.9.7
pynetbox 6.6.2-3
Postgresql 13.7
Nginx being used as proxy with self signed cert

Trying to get ansible to connect to Netbox API with simple playbook:
- name: "Test NetBox modules"
  connection: local
  hosts: localhost
  collections:
   - netbox.netbox
  vars:
    ansible_python_interpreter: /usr/bin/python3
  gather_facts: False

  tasks:
    - name: "Example site: present - Update"
      netbox_site:
        netbox_url: https://netbox.mrelab.net
        netbox_token: xxxxxxxxxxxxxxxxxxxxxxx
        data:
         name: California
        state: "present"
        validate_certs: "false"

"msg": "Failed to establish connection to NetBox API"

The server works fine via my browser.  I have also used Postman to access the Netbox API using my token with success using get devices and my devices were returned so I know the server and API are working.   

### Expected Behavior

Should add a region California to Netbox

### Observed Behavior

The full traceback is:
  File "/tmp/ansible_netbox_site_payload_q81wk95t/ansible_netbox_site_payload.zip/ansible_collections/netbox/netbox/plugins/module_utils/netbox_utils.py", line 680, in _connect_netbox_api
    self.version = nb.version
  File "/usr/local/lib/python3.9/site-packages/pynetbox/core/api.py", line 134, in version
    version = Request(
  File "/usr/local/lib/python3.9/site-packages/pynetbox/core/query.py", line 193, in get_version
    req = self.http_session.get(
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 460, in send
    self.cert_verify(conn, request.url, verify, cert)
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 263, in cert_verify
    raise OSError(
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "cert": null,
            "data": {
                "asn": null,
                "comments": null,
                "contact_email": null,
                "contact_name": null,
                "contact_phone": null,
                "custom_fields": null,
                "description": null,
                "facility": null,
                "latitude": null,
                "longitude": null,
                "name": "California",
                "physical_address": null,
                "region": null,
                "shipping_address": null,
                "site_group": null,
                "slug": null,
                "status": null,
                "tags": null,
                "tenant": null,
                "time_zone": null
            },
            "netbox_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "netbox_url": "https://netbox.mrelab.net",
            "query_params": null,
            "state": "present",
            "validate_certs": "false"
        }
    },
    "msg": "Failed to establish connection to NetBox API"

**Comment by candlerb** [comment]
Aside: your pasted playbook and log show in a garbled fashion. If you edit your post and put three backticks (` ``` `) on a line of their own, before and after each block of code, it should display much better.

An error in `cert_verify` suggests that you don't have a valid certificate for your site - maybe in your browser you just clicked through. You could try `validate_certs: "no"` instead of `validate_certs: "false"`.

If that's not it, try tcpdump or wireshark to look at the traffic to/from Netbox.

**Reply by ellismk** [reply]
Sorry that had no effect. It appears that you can not use self signed certs with netbox.netbox ansible collection only real certs.   So I just ran the nextbox server via:
source /opt/netbox/venv/bin/activate
python3 manage.py runserver 0.0.0.0:8000 --insecure

I then ran my ansible test playbook and it worked.   It added California as a site.  I know it is simple but for now just trying to get things to work.

- name: "Test NetBox modules"
  connection: local
  hosts: localhost
  collections:
   - netbox.netbox
  vars:
    ansible_python_interpreter: /usr/bin/python3
  gather_facts: False

  tasks:
    - name: "Example site: present - Update"
      netbox_site:
        validate_certs: False
        netbox_url: "http://netbox.mrelab.net:8000"
        netbox_token: "xxxxxxxx"
        data:
         name: "California"
         slug: "california"
        state: present

**Reply by rodvand** [reply]
I am using the netbox.netbox collection with self-signed certificates, so it works. `validate_certs: no` should do the trick on the modules. The inventory module requires true/false as value.

**Reply by ellismk** [reply]
Thanks for letting me know.   I had already tried validate_certs: no and False, neither worked.   For now ansible playbooks are working using no certs so that is good for now.

**Comment by ellismk** [comment, marked as answer]
Looks like this is not a Netbox v3.2.5 problem but and issue with the ansible collection netbox.netbox.   Looking at the netbox.netbox discussions they recommend trying setting validate_certs to either no or False, neither works.   I would think if I set validate_certs to either no or False it should work with self signed certs.   For now I have removed my self signed cert in nginx and just using it with no self signed certs.   This allows my ansible playbooks to work.
